File: 1827.最少操作使数组递增/1827-最少操作使数组递增.py
class Solution(object):
    def minOperations(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 1:
            return 0 
        res = 0
        for i, num in enumerate(nums):
            if i:
                if num <= nums[i - 1]:
                    res += (nums[i - 1] + 1) - num
                    nums[i] = nums[i - 1] + 1
            # print(nums)
        return res
                

File: 1827.最少操作使数组递增/1827-最少操作使数组递增.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
    	int res = 0;
    	for (int i = 1; i < nums.size(); i++)
    	{
    		if (nums[i] <= nums[i - 1]) {
    			res += nums[i - 1] - nums[i] + 1; 
                nums[i] = nums[i - 1] + 1;
    		}
    	}
    	return res;
    }
};
